### PR TITLE
[Task 1855214] gradle messager adjustment

### PR DESCRIPTION
**Pull request by andxu**
fix some bugs in graldle plugin :
1. adjust gradle messager to fix the interrupted log caused by system.out.println 
2. adjust the prompt message in proxy
3. adjust the task dependency according to docker 
4. adjust the service plan resource group and name logic which are handled in app-service-lib
5. unify exception properties in webapp gradle plugin telemetry

### CODE REVIEW

**Comment by andxu** [comment]
AB#1859714

**Comment by andxu** [comment]
AB#1855214

File: azure-gradle-plugins-common/src/main/java/com/microsoft/azure/gradle/util/GradleProxyUtils.java [hunk 1]
```diff
@@ -15,8 +15,9 @@
     @SuppressWarnings("HttpUrlsUsage")
     public static void configureProxy() {
         final ProxyManager proxyManager = ProxyManager.getInstance();
+        ProxyManager.getInstance().init();
```
Comment (shenqianjin) [inline azure-gradle-plugins-common/src/main/java/com/microsoft/azure/gradle/util/GradleProxyUtils.java:18]: logger messager

**Review by wangmingliang-ms — APPROVED** [review]

File: azure-gradle-plugins-common/src/main/java/com/microsoft/azure/gradle/util/GradleAzureMessager.java [hunk 2]
```diff
@@ -8,41 +8,37 @@
...
-        } else {
-            System.out.println(message);
-        }
-    }
-
     @Override
-    public String value(String val) {
-        return TextUtils.cyan(val);
+    public String decorateValue(@Nonnull Object p, @Nullable IAzureMessage message) {
+        return TextUtils.cyan(p.toString());
```
Comment (Flanker32) [inline azure-gradle-plugins-common/src/main/java/com/microsoft/azure/gradle/util/GradleAzureMessager.java:42]: Shall we add a default impl for `decorateValue`? It seems that only IDE have different behavior

Comment (andxu) [reply azure-gradle-plugins-common/src/main/java/com/microsoft/azure/gradle/util/GradleAzureMessager.java:42]: assign miller to impl it in interface 'IAzureMessage.ValueDecorator', @wangmingliang-ms

File: azure-webapp-gradle-plugin/src/main/java/com/microsoft/azure/plugin/webapps/gradle/AzureWebappPlugin.java [hunk 3]
```diff
@@ -49,10 +53,16 @@ public void apply(final Project project) {
             final TaskProvider<Task> jarTask = projectAfterEvaluation.getTasks().named("jar");
 
             deployTask.configure(task -> {
-                TaskProvider<Task> targetTask = ObjectUtils.firstNonNull(bootWarTask, bootJarTask, warTask, jarTask);
-                task.dependsOn(targetTask);
-                task.setArtifactFile(Optional.ofNullable(targetTask)
-                    .map(Provider::get).map(Task::getOutputs).map(TaskOutputs::getFiles).map(FileCollection::getAsPath).orElse(null));
+                boolean isDocker = extension.getRuntime() != null && OperatingSystem.fromString(extension.getRuntime().os()) == OperatingSystem.DOCKER;
```
Comment (Flanker32) [inline azure-webapp-gradle-plugin/src/main/java/com/microsoft/azure/plugin/webapps/gradle/AzureWebappPlugin.java:56]: isDocker = Optional.ofNullable(extension.getRuntime()).map(runtime -> OperatingSystem.fromString(runtime os()) == OperatingSystem.DOCKER || (StringUtils.isEmpty(runtime.os() && StringUtils.isNotEmpty(runtime.image()))))

**Review by Flanker32 — APPROVED** [review]

**Pull request merged**